packages/flame_riverpod/lib/src/widget.dart

```diff
@@ -0,0 +1,206 @@
+import 'package:flame/game.dart';
+import 'package:flame_riverpod/flame_riverpod.dart';
+import 'package:flutter/widgets.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+class RiverpodAwareGameWidget<T extends Game> extends GameWidget<T> {
```

Comment:
Some dartdocs here would be good

Reply:
Added some comments, I'll leave it to you to resolve if this is satisfactory

Reply:
Looks great!